frontend/packages/console-app/src/actions/hooks/useCommonActions.ts

```diff
@@ -73,15 +74,24 @@ export const useCommonActions = <T extends readonly CommonActionCreator[]>(
           }),
         accessReview: asAccessReview(kind, resource, 'delete'),
       }),
-      [CommonActionCreator.Edit]: (): Action => ({
-        id: `edit-resource`,
-        label: t('console-app~Edit {{kind}}', { kind: kind.kind }),
-        cta: {
-          href: `${resourceObjPath(resource, kind.crd ? referenceForModel(kind) : kind.kind)}/yaml`,
-        },
-        // TODO: Fallback to "View YAML"? We might want a similar fallback for annotations, labels, etc.
-        accessReview: asAccessReview(kind, resource, 'update'),
-      }),
+      [CommonActionCreator.Edit]: (): Action => {
+        let href: string;
+        switch (kind.kind) {
+          case SecretModel.kind:
+            href = `${resourceObjPath(resource, kind.kind)}/edit`;
+            break;
+          default:
+            href = `${resourceObjPath(resource, kind.crd ? referenceFor(kind) : kind.kind)}/yaml`;
+        }
+        return {
+          id: `edit-resource`,
```

Comment:
Nit - we should avoid `` ` `` template literals if there are no `${x}` expressions.

```suggestion
          id: 'edit-resource',
```